### PR TITLE
Ophys stim epochs and fov

### PR DESCRIPTION
**Pull request by arielleleon**
This contains a huge amount of changes to create the session.json for the mesoscope platforms. 

- Pull metadata for FieldOfView generation from tiff header and hdf5 files
- Index FieldOfView objects correctly
- Correctly add coupled_fov_index
- Parse platform.json for new power fields
- Create stim_epochs
- Add session_types
- Create data_description.json to save project codes in the summary attribute
- Remodel the camstim module so the parent class generates visual stimuli
- Add unit tests for camstim module
- Add unit tests for mesoscope
- Add resources for testing mesoscope session.json

@mekhlakapoor and @rcpeene : I did not unit test the OpenEphysCamstim object but I am guessing it can follow what I did for the parent Camstim class. It also needs to be tested to make sure that the opto files generate properly.

@Ahad-Allen - Will you please verify the camstim tests are producing values you would expect from the methods that build both the stim tables and epochs

### CODE REVIEW

**Comment by rcpeene** [comment]
Thanks for being careful in pulling out all the CamstimEphys methods. I basically had CamstimEphysSessionEtl absorb that OpenEphys class you made because I don't think it has any other distinct uses, if that's alright.

I tested and it works for me!

**Review by rcpeene — APPROVED** [review]

**Review by Ahad-Allen — DISMISSED** [review]
Tested and seems to be good to go! Note that some test cases need linting in their mocks to avoid failing due to lines being too long

**Review by rcpeene — APPROVED** [review]

**Comment by arielleleon** [comment]
@mekhlakapoor - Can you help with linting. I cannot find the lines in tests\test_stimulus\camstim.py that flak8 is complaining about :/

File: src/aind_metadata_mapper/mesoscope/models.py [hunk 1]
```diff
@@ -2,29 +2,54 @@
...
+    )
+    mouse_platform_name: str = Field(
+        default="disc", title="Name of the mouse platform"
     )
-    mouse_platform_name: str = "disc"
+    optional_output: Optional[Path] = Field(
+        default=None, title="Optional output path"
+    )
+
+    @field_validator("input_source", "behavior_source", "output_directory")
```
Comment (mekhlakapoor) [inline src/aind_metadata_mapper/mesoscope/models.py:49]: Can we remove this field_validator? It's empty

**Review by mekhlakapoor — APPROVED** [review]

**Pull request merged**